api/src/main/java/org/openmrs/module/ips/api/impl/InternationalPatientSummaryServiceImpl.java

```diff
@@ -113,26 +119,48 @@ public void addPatientIPS(String uuid) throws Exception {
 						"\"valueIdentifier\": { " +
 						"\"value\": \"%s\" " +
 						"} }] }",
-				uuid);
+				patientID);
 
 		HttpEntity<String> request = new HttpEntity<>(requestBody, headers);
 		ResponseEntity<String> response = restTemplate.postForEntity(url, request, String.class);
 
-		UUID randomUuid = UUID.randomUUID();
+		Obs obs = getIPSObs(uuid);
+
+		String clobUuid;
+		ClobDatatypeStorage clobData;
+
+		if (obs != null && obs.getValueComplex() != null) {
+			clobUuid = obs.getValueComplex();
+			clobData = datatypeService.getClobDatatypeStorageByUuid(clobUuid) != null
+					? datatypeService.getClobDatatypeStorageByUuid(clobUuid)
+					: new ClobDatatypeStorage();
+		} else {
+			clobUuid = UUID.randomUUID().toString();
+			clobData = new ClobDatatypeStorage();
+			clobData.setUuid(clobUuid);
+		}
 
-		ClobDatatypeStorage clobData = new ClobDatatypeStorage();
-		clobData.setUuid(randomUuid.toString());
 		clobData.setValue(response.getBody());
 		datatypeService.saveClobDatatypeStorage(clobData);
 
-		Obs obs = new Obs();
-		String ipsconcept = Context.getAdministrationService()
-				.getGlobalProperty(InternationalPatientSummaryConstants.IPS_CONCEPT);
-		obs.setConcept(conceptService.getConceptByReference(ipsconcept));
-		obs.setValueComplex(randomUuid.toString());
-		obs.setObsDatetime(new Date());
-		obs.setPerson(personService.getPersonByUuid(uuid));
-		obsService.saveObs(obs, "ipsObs");
+		if (obs == null) {
+			obs = new Obs();
+			ipsconcept = administrationService.getGlobalProperty(InternationalPatientSummaryConstants.IPS_CONCEPT);
+			obs.setConcept(conceptService.getConceptByReference(ipsconcept));
+			obs.setValueComplex(clobUuid);
+			obs.setObsDatetime(new Date());
+			obs.setPerson(personService.getPersonByUuid(uuid));
+			obsService.saveObs(obs, "ipsObs");
```

Comment:
Could we make the message here "Create IPS"

Reply:
Also, shouldn't we call `saveObs()` regardless? Otherwise, we're relying on a session flush, which is not a great idea.

Reply:
Done